src/zimdump.cpp

```diff
@@ -393,124 +402,128 @@ void ZimDumper::verifyChecksum()
     std::cout << "no checksum" << std::endl;
 }
 
-int main(int argc, char* argv[])
+static const char USAGE[] =
+R"(zimdump.
+
+    Usage:
+      zimdump <zim_file> info [--ns=N]
+      zimdump <zim_file> list [--details] [--idx=INDEX|(--url=URL [--ns=N])]
+      zimdump <zim_file> dump --dir=DIR [--ns=N] [--redirect]
+      zimdump <zim_file> dump (--idx=INDEX |(--url=URL [--ns=N]))
+
+    Options:
+      --idx=INDEX       The index of the article to list/dump
+      --ns=N            Namespace selection [default: A]
+      -h --help         Show this screen.
+      --version         Show version.
+)";
+
+void subcmdInfo(ZimDumper &app, std::map<std::string, docopt::value> &args)
 {
-  try
-  {
-    zim::Arg<bool> fileinfo(argc, argv, 'F');
-    zim::Arg<char> nsinfo(argc, argv, 'N');
-    zim::Arg<bool> info(argc, argv, 'i');
-    zim::Arg<bool> data(argc, argv, 'd');
-    zim::Arg<bool> page(argc, argv, 'p');
-    zim::Arg<const char*> find(argc, argv, 'f');
-    zim::Arg<const char*> url(argc, argv, 'u');
-    zim::Arg<bool> list(argc, argv, 'l');
-    zim::Arg<bool> tableList(argc, argv, 'L');
-    zim::Arg<zim::size_type> indexOffset(argc, argv, 'o');
-    zim::Arg<bool> extra(argc, argv, 'x');
-    zim::Arg<char> ns(argc, argv, 'n', 'A');  // namespace
-    zim::Arg<const char*> dumpAll(argc, argv, 'D');
-    zim::Arg<bool> verbose(argc, argv, 'v');
-    zim::Arg<bool> titleSort(argc, argv, 't');
-    zim::Arg<bool> verifyChecksum(argc, argv, 'C');
-    zim::Arg<bool> printVersion(argc, argv, 'V');
-#ifndef _WIN32
-    zim::Arg<bool> redirectSymlink(argc, argv, 's');
+    if (args["--namespaceinfo"]) {
+        app.printNsInfo(args["--namespaceinfo"].asString().at(0));
+    }
+    else {
+        app.printInfo();
+    }
+}
+
+void subcmdDumpAll(ZimDumper &app, std::map<std::string, docopt::value> &args, bool redirect, std::function<bool (const char c)> nsfilter)
+{
+#ifdef _WIN32
+    app.dumpFiles(args["--output"].asString(), false, nsfilter);
+#else
+    app.dumpFiles(args["--dir"].asString(), redirect, nsfilter);
```

Comment:
Why we use `--output` on WIN32 and `--dir` on linux/mac ?

Reply:
My type. Thanks